Swift/BRCryptoDemo/CoreDemoListener.swift

```diff
@@ -94,21 +99,29 @@ class CoreDemoListener: SystemListener {
             // specifically, test networks are announced and having a wallet manager for a
             // testnet won't happen in a deployed App.
 
-            if isMainnet == network.isMainnet &&
-                currencyCodesNeeded.contains (where: { nil != network.currencyBy (code: $0) }) {
+            if isMainnet == network.isMainnet && network.currencies
+                .contains (where: { (c) in networkCurrencyCodes.contains { c.code == $0 }}) {
+
                 let mode = system.supportsMode (network: network, WalletManagerMode.api_only)
                     ? WalletManagerMode.api_only
                     : system.defaultMode(network: network)
+
                 let scheme = system.defaultAddressScheme(network: network)
 
+                let currencies = network.currencies
+                    .filter { (c) in registerCurrencyCodes.contains { c.code == $0 } }
+
                 let _ = system.createWalletManager (network: network,
                                                     mode: mode,
-                                                    addressScheme: scheme)
+                                                    addressScheme: scheme,
+                                                    currencies: currencies)
```

Comment:
This feels tricky to me; we can't pass in the BTC/BCH/ETH "primary currencies", ya? This two level filtering ("networkCurrencyCodes"/"registerCurrencyCodes") is necessary? Can we just allow the "network currency codes" to work in the `cryptoWalletManagerRegisterWallet` function?

Maybe I am missing something.

Reply:
In the demo listener I use one set of currencyCodes for primary wallets to select the needed wallet manager and to specify the WalletManagerMode (the mode stuff is on some other branch somewhere, pending). The new, other set is used to register non-primary wallets.  

I think one set is okay.  Still need a '[String,WalletManagerMode]' map.